pkg/internal/controller/controller.go

```diff
@@ -123,16 +172,94 @@ func (c *Controller) Watch(src source.Source, evthdler handler.EventHandler, prc
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	watchDesc := &watchDescription{src: src, handler: evthdler, predicates: prct}
+
+	// If the source is cluster aware, store it in a separate list.
+	if watchDesc.IsClusterAware() {
+		c.clusterAwareWatches = append(c.clusterAwareWatches, watchDesc)
+	}
+
 	// Controller hasn't started yet, store the watches locally and return.
 	//
 	// These watches are going to be held on the controller struct until the manager or user calls Start(...).
 	if !c.Started {
-		c.startWatches = append(c.startWatches, watchDescription{src: src, handler: evthdler, predicates: prct})
+		c.startWatches = append(c.startWatches, watchDesc)
 		return nil
 	}
 
+	var errs []error
 	c.LogConstructor(nil).Info("Starting EventSource", "source", src)
-	return src.Start(c.ctx, evthdler, c.Queue, prct...)
+	if err := src.Start(c.ctx, evthdler, c.Queue, prct...); err != nil {
+		errs = append(errs, err)
+	}
+
+	// If the watch is cluster aware, start it for all the clusters
+	// This covers the case where a Watch was added later to the controller.
+	if watchDesc.IsClusterAware() {
+		for _, cldesc := range c.clusters {
+			if err := c.startClusterAwareWatchLocked(cldesc, watchDesc); err != nil {
```

Comment:
Q: Are cluster aware watches started here and in l.192? If yes and it's intentional do we have to ensure that c.clusters does not contain the "default" cluster? Or is that the responsibility of the provider?